### PR TITLE
Add auto-generation of markdown for resource references

### PR DESCRIPTION
**Pull request by AaronCrawfis**
# Description

This PR adds auto-generation of resource reference markdown docs. Specifically it:

- Updates the Bicep generator to output a new markdown table format
- Generates the new docs
- Adds a workflow to run this on every PR/commit and then PR the results into the docs repo on every push

## Type of change

- This pull request fixes a bug in Radius and has an approved issue (issue link required).
- This pull request adds or changes features of Radius and has an approved issue (issue link required).
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Partially addresses #6101 

### Summary
📝🚀📚

Renamed and updated the workflow to publish CLI and resource reference docs for Radius. Enhanced the custom tool to generate Bicep types and Markdown files for each resource type. Deleted the unused docs workflow.

> _To document Bicep types with ease_
> _We added a function to write tables, please_
> _It formats the Markdown_
> _With headings and links down_
> _And sorts the resources like cheese_

### Walkthrough
*  Rename job and add comment to publish-docs workflow ([link](https://github.com/radius-project/radius/pull/6331/files?diff=unified&w=0#diff-69250281a3f8b6ed9c93d9d9d8d92aa49e2105c37fe83fd22168ec5ab85250bdL16-R16), [link](https://github.com/radius-project/radius/pull/6331/files?diff=unified&w=0#diff-69250281a3f8b6ed9c93d9d9d8d92aa49e2105c37fe83fd22168ec5ab85250bdR49))
*  Add steps to generate and upload resource reference documentation as an artifact ([link](https://github.com/radius-project/radius/pull/6331/files?diff=unified&w=0#diff-69250281a3f8b6ed9c93d9d9d8d92aa49e2105c37fe83fd22168ec5ab85250bdR60-R74))
*  Import `orderBy` function from `lodash` library in `main.ts` ([link](https://github.com/radius-project/radius/pull/6331/files?diff=unified&w=0#diff-633e7d47bcee2f2a8db8b422502805e8e1fecfd165303507820a102c60b3e58dL19-R24))
*  Iterate over resource types and write Markdown files for each one in `main.ts` ([link](https://github.com/radius-project/radius/pull/6331/files?diff=unified&w=0#diff-633e7d47bcee2f2a8db8b422502805e8e1fecfd165303507820a102c60b3e58dR47-R54))
*  Create `markdown-table.ts` file to write Markdown tables for resource types and their properties ([link](https://github.com/radius-project/radius/pull/6331/files?diff=unified&w=0#diff-d614ef1f65b05b6a6171e4284f3a85969175eb18293cea68cb9c27e86215216eL1-R258))
*  Delete unused `docs.yaml` workflow file ([link](https://github.com/radius-project/radius/pull/6331/files?diff=unified&w=0#diff-d54d69dbb27e75dae25cb4b2384310cb57707e419377cf572d5cb0ecc1f16877))

### CODE REVIEW

**Review by rynowak — DISMISSED** [review]

File: .github/workflows/publish-docs.yaml [hunk 1]
```diff
@@ -41,6 +41,15 @@ jobs:
           path: docs
           ref: ${{ env.DOCS_BRANCH }}
           token: ${{ secrets.GH_RAD_CI_BOT_PAT }}
+      # Generate resource reference docs
+      - name: Generate resource reference docs
+        run: python docs/.github/scripts/generate_resource_references.py ./radius/hack/bicep-types-radius/generated/ ./docs/docs/content/reference/resources
```
Comment (rynowak) [inline .github/workflows/publish-docs.yaml:46]: This needs to be hooked up to `make generate`. If the files are checked in, then they need to be updated as part of the PR process.

Comment (rynowak) [reply .github/workflows/publish-docs.yaml:46]: Ideally this would run a `make` target.

Comment (AaronCrawfis) [reply .github/workflows/publish-docs.yaml:40]: this has been addressed after our offline discussion

**Review by rynowak — APPROVED** [review]

**Pull request merged**